bin/lsif_indexer.dart

```diff
@@ -28,15 +28,34 @@
 // Licensed under the BSD-2 Clause License: https://github.com/astashov/crossdart/blob/master/LICENSE
 
 import 'dart:io';
+import 'package:path/path.dart';
+
 import 'package:lsif_indexer/analyzer.dart';
+import 'package:lsif_indexer/src/arguments.dart';
+import 'package:lsif_indexer/src/emitter.dart';
+import 'package:lsif_indexer/src/util/path_extensions.dart';
 
-/// Generate LSIF information for the directory listed in the first [argument], or
+/// Generate LSIF information for the directory provided from the [arguments], or
 /// the current directory if not specified.
 ///
-/// Currently prints to standard out.
+/// The format of the output is based on the [arugments].
 void main(List<String> arguments) async {
-  // TODO: Allow specifying an output file.
-  await Analyzer(
-    arguments.isEmpty ? Directory.current.absolute.path : arguments.first,
-  ).analyzePackage();
+  final config = LsifDartArgumentParser().parse(arguments);
+
+  // Exit early if the arguments were invalid
+  if (!config.isValid) return;
+
+  if (config.output != null) {
```

Comment:
beyond nitpicking: I like to use positive tests, and even though null tests is a bit of a special case there, might as well be == null. Or could even be the single-assignment form
`   emitter = config.output == null ? Emitter.standardOutput() : Emitter.file...`